webgl/penworld.js

```diff
@@ -624,3 +624,87 @@ function render () {
 
   window.requestAnimationFrame(render)
 }
+
+Vector3.prototype.listify = function() { // compact
+  return [this.x, this.y, this.z];
+};
+
+Vector3.prototype.fromList = function(l) { // extract
+  this.x = l[0];
+  this.y = l[1];
+  this.z = l[2];
+}
```

Comment:
I would rather you make these functions that take vectors, or move these methods to renderer.js where Vector3 is defined, rather than modifying the prototype. `listify` could be renamed to `toJSON` (so it gets automatically called by `JSON.stringify`), and if not perhaps `serialize`. I think `fromList` should be a static method instead